$MirOS: ports/devel/cvs/patches/patch-src_server_c,v 1.2 2010/09/15 20:57:03 tg Exp $
--- src/server.c.orig	Wed Sep 28 15:25:37 2005
+++ src/server.c	Wed Sep 15 23:33:02 2010
@@ -20,6 +20,8 @@
 #include "getline.h"
 #include "getnline.h"
 
+__RCSID("$MirOS: ports/devel/cvs/patches/patch-src_server_c,v 1.2 2010/09/15 20:57:03 tg Exp $");
+
 int server_active = 0;
 
 #if defined (SERVER_SUPPORT) || defined (CLIENT_SUPPORT)
@@ -799,6 +801,14 @@ serve_root (char *arg)
 	return;
     }
 
+    if (root_allow_used() && !root_allow_ok(arg))
+    {
+	if (alloc_pending (80 + strlen (arg)))
+	    sprintf (pending_error_text,
+		     "E Bad root %s", arg);
+	return;
+    }
+
     /* Set original_parsed_root here, not because it can be changed in the
      * client Redirect sense, but so we don't have to switch in code that
      * runs in both modes to decide which to print.
@@ -2350,6 +2360,7 @@ prepost_proxy_proc (const char *reposito
     bool *pre = closure;
 
     /* %c = cvs_cmd_name
+     * %I = commit ID
      * %p = shortrepos
      * %r = repository
      */
@@ -2367,6 +2378,7 @@ prepost_proxy_proc (const char *reposito
 # endif /* SUPPORT_OLD_INFO_FMT_STRINGS */
 	                      filter,
 	                      "c", "s", cvs_cmd_name,
+			      "I", "s", global_session_id,
 	                      "R", "s", referrer ? referrer->original : "NONE",
 	                      "p", "s", ".",
 	                      "r", "s", current_parsed_root->directory,
@@ -6887,7 +6899,7 @@ cleanup:
 static int
 check_pam_password (char **username, char *password)
 {
-    int retval, err;
+    int retval;
     struct pam_conv conv = { cvs_pam_conv, 0 };
     char *pam_stage = "start";
 
@@ -6930,7 +6942,7 @@ check_pam_password (char **username, cha
 
     return retval == PAM_SUCCESS;       /* indicate success */
 }
-#endif
+#else /* !HAVE_PAM */
 
 static int
 check_system_password (char *username, char *password)
@@ -6996,6 +7008,7 @@ error 0 %s: no such user\n", username);
 #endif
     return 1;
 }
+#endif /* !HAVE_PAM */
 
 
 
